### PR TITLE
Made naming of some FilterChainOptions instances more consistent

### PR DESCRIPTION
**Pull request by DvoryankinEvgeny**
Commit Message: Made naming of some FilterChainOptions instances more consistent
Additional Description: This PR brings cosmetic changes discussed in https://github.com/envoyproxy/envoy/pull/35292#discussion_r1693184337
Risk Level: Low
Testing: Not required
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: test/extensions/filters/network/http_connection_manager/config_filter_chain_test.cc [hunk 1]
```diff
@@ -294,7 +294,7 @@ TEST_F(FilterChainTest, CreateCustomUpgradeFilterChain) {
                                      filter_config_provider_manager_, creation_status_);
   ASSERT_TRUE(creation_status_.ok());
 
-  Http::EmptyFilterChainOptions option;
+  Http::EmptyFilterChainOptions options;
```
Comment (botengyao) [inline test/extensions/filters/network/http_connection_manager/config_filter_chain_test.cc:297]: nit: const, and others as well.

**Review by botengyao — APPROVED** [review]
LGTM, and thanks for the clean up!

**Comment by DvoryankinEvgeny** [comment]
@botengyao @KBaichoo Hi,
Seems this PR stuck a little bit, so I decided to take a chance and fix the only comment :) Hope it will help this PR to be merged.
Thank you

**Review by KBaichoo — APPROVED** [review]
Thank you @DvoryankinEvgeny ! I think it was waiting for the minornit boteng pointed out, but now that that is resolved, it LGTM.

**Pull request merged**